sdk/core/core-http/lib/credentials/tokenCredential.ts

```diff
@@ -11,11 +11,37 @@ export interface TokenCredential {
    * Gets the token provided by this credential.
    *
    * @param scopes The list of scopes for which the token will have access.
-   * @param requestOptions The RequestOptionsBase used to configure any requests
-   *                       this TokenCredential implementation might make.
+   * @param getTokenOptions The options used to configure any requests
+   *                        this TokenCredential implementation might make.
    */
   getToken(
     scopes: string | string[],
-    requestOptions?: RequestOptionsBase
-  ): Promise<string | null>;
+    getTokenOptions?: GetTokenOptions
```

Comment:
When @schaabs first suggested `GetTokenOptions` I kept reading it as "get tokenoptions" instead of "gettoken options" and was wondering why is he suggesting `GetTokenOptions` as an interface rather than a method name 🤣 and how would this method be useful to anyone!!

Reply:
You're right, it does sound pretty ambiguous when spoken out loud!